graph/telemetry/istio/istio.go

```diff
@@ -966,3 +1008,36 @@ func promQuery(query string, queryTime time.Time, api prom_v1.API) model.Vector
 
 	return nil
 }
+
+// hasWaypoint returns true if the source or dest workload is determined to be a waypoint workload.  Note that this logic can
+// go away if and when https://github.com/istio/ztunnel/issues/1128 is implemented, and then we can make this determination
+// directly from the telemetry
+func hasWaypoint(ztunnel bool, sourceCluster, sourceWlNs, srcWl, destCluster, destWlNs, destWl string, globalInfo *graph.GlobalInfo) (sourceIsWaypoint bool, destIsWaypoint bool) {
+	if !ztunnel {
+		return false, false
+	}
+
+	// first, try a simple substring check on the name, for a common waypoint giveaway
+	sourceIsWaypoint = strings.Contains(strings.ToLower(srcWl), "waypoint")
```

Comment:
I know this is an optimization but there is a small possibility of having a node containing waypoint but not being a waypoint?

Reply:
I agree there is a very small risk.  Originally, I thought it was worth the quick test, but I'll just remove it and use the slightly longer and safer approach.

Reply:
Funny, removing it revealed a problem in my new test setup :-)